VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+    RD_KEY_FILE = '/mnt/azure_bek_disk/LinuxPassPhraseFileName'
+    RD_KEY_FILE_MOUNT_POINT = '/mnt/azure_bek_disk'
+    RD_KEY_VOLUME_LABEL = 'BEK VOLUME'
+
+    def __init__(self, hutil, logger, distro_patcher):
+        self.hutil = hutil
+        self.logger = logger
+        self.executor = CommandExecutor(self.logger)
+        self.disk_util = DiskUtil(hutil=self.hutil, patching=distro_patcher, logger=self.logger, encryption_environment=None)
+        self.mapper_name = str(uuid.uuid4())
+        self.mapper_path = self.DM_PREFIX + self.mapper_name
+
+    def is_encrypt_format_all(self):
+        """ return true if current encryption operation is EncryptFormatAll """
+        try:                
+            public_settings_str = self.hutil._context._config['runtimeSettings'][0]['handlerSettings'].get('publicSettings')
+            if isinstance(public_settings_str, basestring):
+                public_settings = json.loads(public_settings_str)
+            else:
+                public_settings = public_settings_str
+            encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
+            if encryption_operation in [CommonVariables.EnableEncryptionFormatAll]:
+                return True
+        except:
+            self.logger.log("unable to identify current encryption operation")
+        return False
+
+    def is_luks_device(self):
+        """ checks if the device is set up with a luks header """
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup isLuks ' + self.RD_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def is_luks_device_opened(self):
+        """ check for presence of luks uuid to see if device was already opened """
+        # suppress logging to avoid log clutter if the device is not open yet
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'test -b /dev/disk/by-uuid/$(cryptsetup luksUUID ' + self.RD_DEV_PATH + ')'
+        return (int)(self.executor.ExecuteInBash(cmd, suppress_logging=True)) == 0
+
+    def is_valid_key(self):
+        """ test if current key can be used to open current partition """
+        # suppress logging to avoid log clutter if the key doesn't match
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup luksOpen ' + self.RD_DEV_PATH + ' --test-passphrase --key-file ' + self.RD_KEY_FILE
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_exists(self):
+        """ true if the udev name for resourced disk exists """
+        cmd = 'test -b ' + self.RD_BASE_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_partition_exists(self):
+        """ true if udev name for resource disk partition exists """
+        cmd = 'test -b ' + self.RD_DEV_PATH
```

Comment:
Maybe there is a better way to do this.. you may want to combine the following two stack overflow solutions and make it a different disk_util function called "is_block_device".
https://stackoverflow.com/questions/82831/how-do-i-check-whether-a-file-exists-using-python
https://stackoverflow.com/questions/32682477/check-if-file-path-is-block-device-in-python

Reply:
Nice, investigating...

Reply:
There may be a better way but I am not seeing it yet even after reviewing these good links.  The test -b seems rather universal without requiring an additional library, and compatible with udev name ( /dev/disk/azure/resource-part1 in this case)